### PR TITLE
Allen catalogue

### PR DESCRIPTION
**Pull request by thorstenhater**
Replaces #1069

### CODE REVIEW

File: mechanisms/generate_catalogue [hunk 1]
```diff
@@ -77,7 +83,7 @@ def parse_arguments():
     return vars(parser.parse_args())
 
 
-def generate(modpfx='', arbpfx='', modules=[], backends=[], **rest):
+def generate(modpfx='', arbpfx='', modules=[], backends=[], catalogue='default', **rest):
```
Comment (bcumming) [inline mechanisms/generate_catalogue:86]: Should we make the catalogue name default `'default'`, or force the user to always specify a name.
My preference is to not have a default name.

Comment (thorstenhater) [reply mechanisms/generate_catalogue:86]: Agreed

**Comment by thorstenhater** [comment]
@noraabiakar yes, fixed

**Comment by thorstenhater** [comment]
bors try

**Pull request merged**